packages/patternfly-4/react-table/src/components/Table/Body.js

```diff
@@ -32,50 +33,33 @@ class ContextBody extends React.Component {
     };
   };
 
-  parentsExpanded(parentId) {
-    const { rows } = this.props;
-    return rows[parentId].hasOwnProperty('parent')
-      ? this.parentsExpanded(rows[parentId].parent)
-      : rows[parentId].isOpen;
-  }
+  mapCells = (headerData, row, rowKey) => {
+    // column indexes start after generated optional columns
+    let additionalColsIndexShift = headerData[0].extraParams.firstUserColumnIndex;
 
-  mapCells = (row, rowKey) => {
-    const { headerData } = this.props;
-    let shiftKey = Boolean(headerData[0] && headerData[0].extraParams.onSelect);
-    shiftKey += Boolean(headerData[0] && headerData[0].extraParams.onCollapse);
     return {
       ...(row &&
         (row.cells || row).reduce(
-          (acc, curr, key) => {
-            const currShift = shiftKey;
-            if (curr) {
-              if (curr.props && curr.props.colSpan) {
-                shiftKey += shiftKey + curr.props && curr.props.colSpan - 1;
-              }
-              return {
-                ...acc,
-                ...{
-                  [headerData[currShift + key].property]: {
-                    title: curr.title || curr,
-                    props: {
-                      isVisible: true,
-                      ...curr.props
-                    }
-                  }
+          (acc, cell, cellIndex) => {
+            const isCellObject = cell === Object(cell);
+
+            const mappedCell = {
+              [headerData[cellIndex + additionalColsIndexShift].property]: {
+                title: isCellObject ? cell.title : cell,
+                props: {
+                  isVisible: true,
+                  ...(isCellObject ? cell.props : null)
                 }
-              };
+              }
+            };
+
+            // increment the shift index when a cell spans multiple columns
+            if (isCellObject && cell.props && cell.props.colSpan) {
+              additionalColsIndexShift += cell.props.colSpan - 1;
             }
-            console.warn(`Cell value at [ ${rowKey}, ${key} ] should not be set to ${curr}`);
             return {
               ...acc,
-              ...{
```

Comment:
This is needed since there are times when curr is set to null or undefined causing the application to crash.  The warning is something the PF4 team decided was appropriate for this situation.